### ISSUE TITLE
Use `ScopedArray` in `MutDeque`

### ISSUE THREAD

**Issue by magnus-madsen** [flix/flix#3362]

**Comment by magnus-madsen** [comment]
@jaschdoc Possibly this one is easier than `MutMap` because it does not have dependencies.

**Comment by jaschdoc** [comment]
Should the enum definition be changed to
```flix
pub enum MutDeque[a: Type, r: Region] {
    case MutDeque(ScopedArray[a, r], ScopedRef[Int32, r], ScopedRef[Int32, r])
}
```
or should there be a `ScopedRef` on the `ScopedArray` or something else entirely? Seems weird to wrap a scopedarray in a scopedref.

**Comment by magnus-madsen** [comment]
Your proposal is correct.

**Comment by magnus-madsen** [comment]
Yes, sorry

You need a ScopedRef of a ScopedArray.

**Comment by jaschdoc** [comment]
One more thing: I think we have to fix #3363 before fixing this since using a ScopedArray breaks all calls to array functions.

**Comment by magnus-madsen** [comment]
> One more thing: I think we have to fix #3363 before fixing this since using a ScopedArray breaks all calls to array functions. Or am I being thrown off by the errors I'm seeing? 😂

Yes, probably. Any use of Array.X now would be impure.
